libs/core/src/lib/micro-process-flow/components/micro-process-flow-item/micro-process-flow-item.component.ts

```diff
@@ -0,0 +1,42 @@
+import { ChangeDetectionStrategy, Component, OnInit, ViewEncapsulation, Input, ChangeDetectorRef, ElementRef } from '@angular/core';
+import { MicroProcessFlowItemType } from '../../types';
+
+@Component({
+  selector: 'fd-micro-process-flow-item',
+  templateUrl: './micro-process-flow-item.component.html',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+      class: 'fd-micro-process-flow__item',
+      '[class.fd-micro-process-flow__item--positive]': 'state === "positive"',
+      '[class.fd-micro-process-flow__item--critical]': 'state === "critical"',
+      '[class.fd-micro-process-flow__item--negative]': 'state === "negative"',
+      '[class.fd-micro-process-flow__item--information]': 'state === "information"',
+      '[class.fd-micro-process-flow__item--last]': '_finalStep === true'
+  }
+})
+export class MicroProcessFlowItemComponent implements OnInit {
+
+    @Input()
+    state: MicroProcessFlowItemType = 'none';
+
+    @Input()
+    intermediate = false;
+
+    _finalStep = false;
+
+    /** @hidden */
+    constructor(
+        private _cd: ChangeDetectorRef,
+        public elRef: ElementRef<HTMLElement>
+    ) { }
+
+    ngOnInit(): void {
+    }
+
+    setFinalStep(value: boolean): void {
+        this._finalStep = value;
+        this._cd.detectChanges();
+    }
```

Comment:
I noticed that this method is invoke only inside loop of parent component. I think it would be better for performance remove  detectChanges from here and invoke it inside parent after the loop.